src/rhsmlib/facts/host_collector.py

```diff
@@ -68,8 +71,16 @@ def get_all(self):
 
         locale_info = {}
         effective_locale = 'Unknown'
-        if locale.getdefaultlocale()[0] is not None:
-            effective_locale = ".".join(filter(None, locale.getdefaultlocale()))
+        # When there is no locale set (system variable LANG is unset),
+        # then this is value returned by locale.getdefaultlocale()
+        # Tuple contains: (language[_territory], encoding identifier)
+        default_locale = (None, None)
+        try:
+            default_locale = locale.getdefaultlocale()
+        except ValueError as err:
+            log.warning("Unable to get default locale (wrong system variable LANG): %s" % err)
```

Comment:
It's not necessarily `LANG`. It can also be `LC_ALL` or others. See https://docs.python.org/2/library/locale.html#locale.getdefaultlocale . So you should just say this instead: `"Unable to get default locale (bad environment variable?): %s"`.